control-plane/grpc/proto/v1/misc/common.proto

```diff
@@ -0,0 +1,86 @@
+syntax = "proto3";
+
+package v1.common;
+
+// Enum defining the various statuses that a resource spec can be in.
+enum SpecStatus {
+  Creating = 0;
+  Created = 1;
+  Deleting = 2;
+  Deleted = 3;
+}
+
+// StringMapValue is the type having a hashmap
+message StringMapValue {
+  map<string, string> value = 1;
+}
+
+// ReplyError to be used for all error propagation to and from grpc calls
+message ReplyError {
+  ReplyErrorKind kind = 1;
+  ResourceKind resource = 2;
+  string source = 3;
+  string extra = 4;
+}
+
+// ReplyErrorKind for the kind of the occured error
+enum ReplyErrorKind {
+  WithMessage = 0;
+  DeserializeReq = 1;
+  Internal = 2;
+  Timeout = 3;
+  InvalidArgument = 4;
+  DeadlineExceeded = 5;
+  NotFound = 6;
+  AlreadyExists = 7;
+  PermissionDenied = 8;
+  ResourceExhausted = 9;
+  FailedPrecondition = 10;
+  Aborted = 11;
+  OutOfRange = 12;
+  Unimplemented = 13;
+  Unavailable = 14;
+  Unauthenticated = 15;
+  Unauthorized = 16;
+  Conflict = 17;
+  FailedPersist = 18;
+  NotShared = 19;
+  AlreadyShared = 20;
+  NotPublished = 21;
+  AlreadyPublished = 22;
+  Deleting_ = 23;
```

Comment:
If possible can we get rid of the underscore suffix. Maybe we could rename it to `IsDeleting`?